xpublish/utils/zarr.py

```diff
@@ -104,10 +108,11 @@ def create_zmetadata(dataset):
     zmeta['metadata'][group_meta_key] = {'zarr_format': zarr_format}
     zmeta['metadata'][attrs_key] = _extract_dataset_zattrs(dataset)
 
-    for key, da in dataset.variables.items():
+    for key in dataset.variables.keys():
+        da = dataset[key]
         encoded_da = encode_zarr_variable(da, name=key)
         encoding = extract_zarr_variable_encoding(da)
-        zmeta['metadata'][f'{key}/{attrs_key}'] = _extract_dataarray_zattrs(encoded_da)
+        zmeta['metadata'][f'{key}/{attrs_key}'] = _extract_dataarray_zattrs(da)
```

Comment:
`encoded_da` encodes the da as a `Variable`, so using `encoded_da` doesn't include the necessary coordinate info. Not sure if there is a better way to deal with this, or if its better to extract the coordinate encoding to its own function so the rest of the attrs use the encoded version